PiwikTracker/Logger.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+public enum LogLevel: Int {
+    case verbose = 10
+    case debug = 20
+    case info = 30
+    case warning = 40
+    case error = 50
+    
+    var shortcut: String {
+        switch self {
+        case .error: return "E"
+        case .warning: return "W"
+        case .info: return "I"
+        case .debug: return "D"
+        case .verbose: return "V"
+        }
+    }
+}
+
+public protocol Logger {
+    var minLevel: LogLevel { get set }
+    func log(message: ()->(String), with level: LogLevel)
```

Comment:
Could capture file, function line number in the logs

````
func log(_ message: Void -> String, with level: LogLevel, file: String = #file, function: String = #function, line: Int = #line )
````

the extension methods below also needs the same params for this to work